src/main/java/org/akaza/openclinica/controller/StudyParticipantController.java

```diff
@@ -505,4 +540,103 @@ public String startBulkAddParticipantJob(MultipartFile file, String schema, Stri
+										           @ApiParam( value = "optional parameter landscape whether paper orientation is landscape. Valid values are true, false. Default is false.", required = false ) @RequestParam(value="landscape",defaultValue = "false",required = false) String landscape,
+										           @Context HttpServletRequest request
+										          ) throws IOException {
+											 
+		
+		  utilService.setSchemaFromStudyOid(studyOid);		 	  
+	 	  UserAccountBean userAccountBean = utilService.getUserAccountFromRequest(request);
+	 	  String siteOid = null;
+	 	  
+		  try {				 
+			  validateService.validateStudyAndRoles(studyOid,userAccountBean);			 	 			 	 		 	 
+		 	  String uuid = startBulkCaseBookPDFJob(studyOid,siteOid, participantId, request, userAccountBean, format, margin, landscape);
+
+			  logger.info("REST request to Casebook PDF Job uuid {} ", uuid);			
+			  return new ResponseEntity<Object>("job uuid: " + uuid, HttpStatus.OK);
+		  
+			  } catch (OpenClinicaSystemException e) {
+					return new ResponseEntity(validateService.getResponseForException(e, studyOid, siteOid), HttpStatus.BAD_REQUEST);
+			}
+		 	 
+		 }
+
+	
+	private String startBulkCaseBookPDFJob(String studyOid,
+											String siteOid, 
+											String participantId, 											 
+											HttpServletRequest request,
+											UserAccountBean userAccountBean, 
+											String format, 
+											String margin, 
+											String landscape) {
+									 	 
+
+		    Study site = null;
+		    Study study = null;
+		    if(siteOid !=null) {
+		    	site = studyDao.findByOcOID(siteOid);
+		    }
+			if(studyOid != null) {
+				study = studyDao.findByOcOID(studyOid);
+			}
+			
+			UserAccount userAccount = uAccountDao.findById(userAccountBean.getId());
+			//currently studySubjectIdentifier is OID
+			StudySubject ss = studySubjectDao.findByLabelAndStudyOrParentStudy(participantId, study);
+			String studySubjectIdentifier = ss.getOcOid();
+			
+			//Setting the destination file
+	        String fullFinalFilePathName = this.getMergedPDFcasebookFileName(studyOid, participantId);
+	        int index= fullFinalFilePathName.lastIndexOf("\\");
+	    	String fileName = fullFinalFilePathName.substring(index + 1);
+	    	
+			JobDetail jobDetail = userService.persistJobCreated(study, site, userAccount, JobType.PARTICIPANT_PDF_CASEBOOK, fileName);
+			jobDetail.setLogPath(fileName);
+			ServletContext servletContext = request.getServletContext();
+			String accessToken = (String) request.getSession().getAttribute("accessToken");
+			servletContext.setAttribute("accessToken", accessToken);
+			servletContext.setAttribute("studyID", study.getStudyId()+"");
+			List<String> permissionTagsString =permissionService.getPermissionTagsList((StudyBean)request.getSession().getAttribute("study"),request);
+			CompletableFuture<Object> future = CompletableFuture.supplyAsync(() -> {
```

Comment:
fetch the locale dynamically by calling LocaleResolver.resolveLocale(request) and then call ResourceBundleProvider.updateLocate from within the Future.